trunk/NDHMS/Routing/Reservoirs/RFC_Forecasts/module_rfc_forecasts.F

```diff
@@ -0,0 +1,129 @@
+! This module defines and instantiates objects
+! for a rfc forecasts persistence levelpool type
+! reservoir. The rfc forecasts reservoir type
+! inherits input and output types from the
+! reservoir base module and calls instantiation
+! of these into sub-objects. The rfc forecasts
+! reservoir type also points to types for
+! rfc forecasts properties and state and calls
+! instantiation of these into sub-objects.
+! There is also a subroutine to run rfc forecasts reservoir
+! that is derived from the reservoir base
+! type interface to run reservoir.
+
+module module_rfc_forecasts
+
+    use module_rfc_forecasts_properties, only: rfc_forecasts_properties_interface
+    use module_rfc_forecasts_state, only: rfc_forecasts_state_interface
+    use module_reservoir, only: reservoir, reservoir_input, reservoir_output
+    implicit none
+
+    ! Extend/derive rfc forecasts type from the abstract base
+    ! type for reservoirs.
+    type, extends(reservoir) :: rfc_forecasts
+
+        ! Define pointers to sub-types / sub-objects to and
+        ! held by a level pool reservoir object.
+        type (rfc_forecasts_properties_interface), pointer :: properties => null()
+        type (rfc_forecasts_state_interface), pointer :: state => null()
+
+        logical :: pointer_allocation_guard = .false.
+
+    contains
+
+        procedure :: init => rfc_forecasts_init
+        procedure :: destroy => rfc_forecasts_destroy
+        procedure :: run => run_rfc_forecasts_reservoir
+
+    end type rfc_forecasts
+
+contains
+
+    ! RFC Forecasts Constructor
+    subroutine rfc_forecasts_init(this, water_elevation,  &
+        lake_area, weir_elevation, weir_coeffecient, &
+        weir_length, orifice_elevation, orifice_coefficient, &
+        orifice_area, lake_max_water_elevation, initial_fractional_depth, &
+        lake_number)
+        implicit none
+        class(rfc_forecasts), intent(inout) :: this ! object being initialized
+        real,    intent(inout) :: water_elevation           ! meters AMSL
+        real,    intent(in)    :: lake_area      		    ! area of lake (km^2)
+        real,    intent(in)    :: weir_elevation            ! bottom of weir elevation (meters AMSL)
+        real,    intent(in)    :: weir_coeffecient          ! weir coefficient
+        real,    intent(in)    :: weir_length               ! weir length (meters)
+        real,    intent(in)    :: orifice_elevation         ! orifice elevation (meters AMSL)
+        real,    intent(in)    :: orifice_coefficient       ! orifice coefficient
+        real,    intent(in)    :: orifice_area              ! orifice area (meters^2)
```

Comment:
This may need to be updated to support the new `dam_length` parameter. Please either a) confirm it doesn't, or b) add support.

Reply:
b) add support.
Thanks

Reply:
@jdmattern-noaa fixing